apps/wallet/src/features/Swap/context.ts

```diff
@@ -0,0 +1,114 @@
+import type { VersionedTransaction } from '@solana/web3.js';
+import type { Networks } from '@walless/core';
+import { AnimateDirections, BindDirections, modalActions } from '@walless/gui';
+import type { TokenDocument } from '@walless/store';
+import { ModalId } from 'modals/types';
+import type { JupiterToken } from 'utils/hooks';
+import type { SwapQuote } from 'utils/transaction';
+import { constructSwapTransaction, getAliasedMint } from 'utils/transaction';
+import { proxy } from 'valtio';
+
+import SelectFromToken from './Select/SelectFromToken';
+import SelectToToken from './Select/SelectToToken';
+import Success from './Success';
+
+export interface SwapContext {
+	network?: Networks;
+	fromToken?: TokenDocument;
+	toToken?: JupiterToken;
+	amount: string;
+	swapQuote?: SwapQuote;
+	transaction?: VersionedTransaction;
+}
+
+const initialContext = {
+	amount: '',
+};
+
+export const swapContext = proxy<{ swap: SwapContext }>({
+	swap: initialContext,
+});
+
+const modalMap = {
```

Comment:
may we make it more "typescript"?

```ts
enum SwapToken {
  From,
  To,
}

interface SwapModal {
  id: string;
  component: ReactNode;
}

const modalMap: Record<SwapToken. SwapModal> = {};
```

Because I see that you have use the `"from"` or `"to`" in other components, we can reuse this `enum`

Reply:
It has been 'typescript' already, typescript allows us to use values as type, in case it needs, a little simple values, I prefer keeping the values as type